### PR TITLE
Feat/32033/show diable qa and tooltip for table custom action

### PR DESCRIPTION
**Pull request by nahmed22**
Disable Add Custom Column action if rows are required and not available

### CODE REVIEW

File: .changeset/fuzzy-onions-exist.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+"@sap-ux-private/control-property-editor-common": patch
+"@sap-ux/control-property-editor": patch
+"@sap-ux-private/preview-middleware-client": patch,
+'@sap-ux/preview-middleware': patch
+---
+
+CPE - Enable Table Filtering  Quick Action
```
Comment (vadson71) [inline .changeset/fuzzy-onions-exist.md:8]: This changeset was added by mistake, I assume

File: packages/preview-middleware-client/src/adp/quick-actions/table-quick-action-base.ts [hunk 2]
```diff
@@ -103,8 +112,11 @@ export abstract class TableQuickActionDefinitionBase {
             if (section) {
                 this.collectChildrenInSection(section, table);
             } else if (this.iconTabBar && tabKey) {
+                let tableRows = this.getInternalTableRows(table);
                 this.children.push({
                     label: `'${iconTabBarfilterMap[tabKey]}' table`,
+                    tooltip: (tableRows.length < 1 && this.tooltip) ? this.context.resourceBundle.getText(this.tooltip) : undefined,
```
Comment (vadson71) [inline packages/preview-middleware-client/src/adp/quick-actions/table-quick-action-base.ts:118]: I don't understand the logic for the `tooltip` and `enabled` props.

File: packages/preview-middleware-client/src/adp/quick-actions/table-quick-action-base.ts [hunk 3]
```diff
@@ -103,8 +112,11 @@ export abstract class TableQuickActionDefinitionBase {
...
+                    enabled: !(tableRows.length < 1 && this.tooltip),
```
Comment (vadson71) [inline packages/preview-middleware-client/src/adp/quick-actions/table-quick-action-base.ts:119]: Why table rows count affects `enabled` property in the base class? It is action specific condition. Why tooltip is considered here?

File: packages/preview-middleware-client/src/adp/quick-actions/table-quick-action-base.ts [hunk 4]
```diff
@@ -131,16 +143,61 @@ export abstract class TableQuickActionDefinitionBase {
...
         }
         if (this.children.length > 0) {
-            this.isActive = true;
+            this.isApplicable = true;
         }
     }
 
+
+    /**
+     * Retrieves the internal table from a UI5Element and checks if it contains rows.
```
Comment (vadson71) [inline packages/preview-middleware-client/src/adp/quick-actions/table-quick-action-base.ts:152]: The function doesn't check anything. It just returns table rows.
I'd suggest to make it a reusable function that returns internal table of the control. 
What to check in this table - it should be business of the function consumer

File: packages/preview-middleware-client/src/adp/quick-actions/table-quick-action-base.ts [hunk 5]
```diff
@@ -212,6 +273,8 @@ export abstract class TableQuickActionDefinitionBase {
                     tableMapIndex = `${tableMapIndex}/${sectionChild.children.length - 1}`;
                     sectionChild.children.push({
                         label: this.getTableLabel(table),
+                        tooltip: (tableRows.length < 1 && this.tooltip) ? this.context.resourceBundle.getText(this.tooltip) : undefined,
+                        enabled: !(tableRows.length < 1 && this.tooltip),
```
Comment (vadson71) [inline packages/preview-middleware-client/src/adp/quick-actions/table-quick-action-base.ts:277]: the same here

File: packages/preview-middleware-client/src/messagebundle.properties [hunk 6]
```diff
@@ -5,12 +5,14 @@ QUICK_ACTION_OP_ADD_CUSTOM_SECTION=Add Custom Section
...
 QUICK_ACTION_ADD_CUSTOM_TABLE_ACTION=Add Custom Table Action
 QUICK_ACTION_ADD_CUSTOM_TABLE_COLUMN=Add Custom Table Column
+QUICK_ACTION_ENABLE_TABLE_FILTERING=Enable Table Filtering for Page Variants
 
 V2_QUICK_ACTION_CHANGE_TABLE_COLUMNS=Change Table Columns
 
 V2_QUICK_ACTION_LR_ENABLE_CLEAR_FILTER_BAR=Enable "Clear" Button in Filter Bar
 V2_QUICK_ACTION_LR_DISABLE_CLEAR_FILTER_BAR=Disable "Clear" Button in Filter Bar
-
+V2_QUICK_ACTION_LR_ENABLE_TABLE_FILTERING=Enable Table Filtering for Page Variants
```
Comment (vadson71) [inline packages/preview-middleware-client/src/messagebundle.properties:14]: Irrelevant

File: packages/preview-middleware-client/src/messagebundle.properties [hunk 7]
```diff
@@ -5,12 +5,14 @@ QUICK_ACTION_OP_ADD_CUSTOM_SECTION=Add Custom Section
 QUICK_ACTION_ADD_CUSTOM_PAGE_ACTION=Add Custom Page Action
 QUICK_ACTION_ADD_CUSTOM_TABLE_ACTION=Add Custom Table Action
 QUICK_ACTION_ADD_CUSTOM_TABLE_COLUMN=Add Custom Table Column
+QUICK_ACTION_ENABLE_TABLE_FILTERING=Enable Table Filtering for Page Variants
```
Comment (vadson71) [inline packages/preview-middleware-client/src/messagebundle.properties:8]: Irrelevant

File: packages/preview-middleware-client/test/unit/adp/quick-actions/fe-v2.test.ts [hunk 8]
```diff
@@ -1098,6 +1111,208 @@ describe('FE V2 quick actions', () => {
                 });
             });
         });
+
+        describe('enable table filtering', () => {
```
Comment (vadson71) [inline packages/preview-middleware-client/test/unit/adp/quick-actions/fe-v2.test.ts:1115]: Irrelevant

File: packages/preview-middleware-client/test/unit/adp/quick-actions/fe-v4.test.ts [hunk 9]
```diff
@@ -903,6 +921,205 @@ describe('FE V4 quick actions', () => {
             });
         });
 
+        describe('enable table filtering', () => {
```
Comment (vadson71) [inline packages/preview-middleware-client/test/unit/adp/quick-actions/fe-v4.test.ts:924]: Irrelevant

**Review by vadson71 — CHANGES_REQUESTED** [review]

**Comment by nahmed22** [comment]
absolete, https://github.com/SAP/open-ux-tools/pull/2622

**Pull request closed without merging**